pkg/imexport/import.go

```diff
@@ -74,9 +74,14 @@ func Untardir(fs vfs.VFS, r io.Reader, dst string) error {
 				}
 			}
 
-			if _, err := io.Copy(file, tr); err != nil {
+			_, err = io.Copy(file, tr)
+			cerr := file.Close()
+			if err != nil {
 				return err
```

Comment:
`file.Close` is called before checking the error from `io.Copy`, so yes, the file will be closed if the copy has failed